src/manage_sql.c

```diff
@@ -21444,30 +21447,60 @@ create_report (array_t *results, const char *task_id, const char *task_name,
         quoted_qod = g_strdup (G_STRINGIFY (QOD_DEFAULT));
       quoted_qod_type = sql_quote (result->qod_type ? result->qod_type : "");
       result_nvt_notice (quoted_nvt_oid);
-      sql ("INSERT INTO results"
-           " (uuid, owner, date, task, host, hostname, port,"
-           "  nvt, type, description,"
-           "  nvt_version, severity, qod, qod_type, result_nvt)"
-           " VALUES"
-           " (make_uuid (), %llu, m_now (), %llu, '%s', '%s', '%s',"
-           "  '%s', '%s', '%s',"
-           "  '%s', '%s', '%s', '%s',"
-           "  (SELECT id FROM result_nvts WHERE nvt = '%s'));",
-           owner,
-           task,
-           quoted_host,
-           quoted_hostname,
-           quoted_port,
-           quoted_nvt_oid,
-           result->threat
-            ? threat_message_type (result->threat)
-            : "Log Message",
-           quoted_description,
-           quoted_scan_nvt_version,
-           quoted_severity,
-           quoted_qod,
-           quoted_qod_type,
-           quoted_nvt_oid);
+
+      if (first)
+        g_string_append (insert,
+                         "INSERT INTO results"
+                         " (uuid, owner, date, task, host, hostname, port,"
+                         "  nvt, type, description,"
+                         "  nvt_version, severity, qod, qod_type, result_nvt,"
+                         "  report)"
+                         " VALUES");
+      else
+        g_string_append (insert, ", ");
+      first = 0;
+      g_string_append_printf (insert,
+                              " (make_uuid (), %llu, m_now (), %llu, '%s',"
+                              "  '%s', '%s', '%s', '%s', '%s', '%s', '%s',"
+                              "  '%s', '%s',"
+                              "  (SELECT id FROM result_nvts WHERE nvt = '%s'),"
+                              "  %llu)",
+                              owner,
+                              task,
+                              quoted_host,
+                              quoted_hostname,
+                              quoted_port,
+                              quoted_nvt_oid,
+                              result->threat
+                               ? threat_message_type (result->threat)
+                               : "Log Message",
+                              quoted_description,
+                              quoted_scan_nvt_version,
+                              quoted_severity,
+                              quoted_qod,
+                              quoted_qod_type,
+                              quoted_nvt_oid,
+                              report);
+
+      /* Insert at most 300 results at a time. */
+      if (insert_count == 300)
```

Comment:
As said in the summary, I'd prefer using a macro here.
Somewhere near CREATE_REPORT_CHUNK_SIZE would probably be a good place to define it.